File: 6kyu/findOutlier.js
function findOutlier(integers) {
    let odd = []
    let even = []
    for (let i in integers) {
        if (integers[i] % 2 === 0) even.push(integers[i])
        else odd.push(integers[i])
    }
    return odd.length > even.length ? +even.join() : +odd.join()
}


console.log(findOutlier([0, 1, 2]));
console.log(findOutlier([1, 2, 3]));
console.log(findOutlier([2, 6, 8, 10, 3]));
console.log(findOutlier([1, 1, 0, 1, 1]));

File: 6kyu/findOutlier.py
def find_outlier(integers):
    odd = []
    even = []
    for i in range(len(integers)):
        if integers[i] % 2 == 0: even.append(integers[i])
        else: odd.append(integers[i])
    return odd[0] if len(odd)< len(even) else even[0]



print(find_outlier([0, 1, 2]));
print(find_outlier([1, 2, 3]));
print(find_outlier([2, 6, 8, 10, 3]));
print(find_outlier([1, 1, 0, 1, 1]));